datapackage_pipelines_assembler/nodes/basic_nodes.py

```diff
@@ -56,3 +56,42 @@ def get_artifacts(self):
             False
         )
         yield output
+
+
+class DerivedPreviewProcessingNode(BaseProcessingNode):
+    def __init__(self, available_artifacts, outputs):
+        super(DerivedPreviewProcessingNode, self).__init__(available_artifacts, outputs)
+        self.fmt = 'json'
+
+    def get_artifacts(self):
+        for artifact in self.available_artifacts:
+            if artifact.datahub_type == 'derived/csv':
+                datahub_type = 'derifdfdved/preview'
+                resource_name = artifact.resource_name + '_preview'
+                output = ProcessingArtifact(
+                    datahub_type, resource_name,
+                    [artifact], [],
+                    [('assembler.update_resource',
+                      {
+                          'name': artifact.resource_name,
+                          'update': {
+                              'name': resource_name,
+                              'format': self.fmt,
+                              'path': 'data/{}.{}'.format(resource_name, self.fmt),
+                              'datahub': {
+                                'type': "derived/preview",
+                                'derivedFrom': [
+                                    artifact.resource_name
```

Comment:
again is this original resource name?

Reply:
No this is the name of the resource that it is derived. Eg: vix-daily_csv

Reply:
No this is name of derived resource in this case `resource-name_csv`